### PR TITLE
🪟 🎨 🧹 Migrate `Input` and `TextArea` to SCSS 

### PR DESCRIPTION
**Pull request by edmundito**
## What
Related to [#15791](https://github.com/airbytehq/airbyte/issues/15791)

Migrates `Input` and `TextArea` to SCSS and adds Storybook. The styles are kept as they were before exactly instead of applying any updates that my mismatch the current Figma design (this may be done as a separate PR in the future.)

## How
* Updates the color names
* Add storybook

## Recommended reading order
Top to bottom

## Tests

* Ensured that disabled, light, error, and password states behave as expected

### CODE REVIEW

File: airbyte-webapp/src/components/base/Input/Input.tsx [hunk 1]
```diff
@@ -137,15 +68,34 @@ const Input: React.FC<InputProps> = ({ ...props }) => {
...
+          {
+            [styles.disabled]: props.disabled,
+            [styles.password]: isPassword,
+          },
+          props.className
+        )}
+      />
       {isVisibilityButtonVisible ? (
-        <VisibilityButton
+        <Button
```
Comment (krishnaglick) [inline airbyte-webapp/src/components/base/Input/Input.tsx:97]: Replacing button looks like it's going to be _fun_

File: airbyte-webapp/src/components/base/TextArea/TextArea.module.scss [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  border-radius: 4px;
+  font-size: 14px;
+  line-height: 20px;
+  font-weight: normal;
+  border: 1px solid colors.$grey-50;
+  background-color: colors.$grey-50;
+  color: colors.$dark-blue;
+  caret-color: colors.$blue;
+
+  .error {
```
Comment (krishnaglick) [inline airbyte-webapp/src/components/base/TextArea/TextArea.module.scss:18]:
```suggestion
  &.error {
```

**Review by krishnaglick — COMMENTED** [review]
Only one issue I found, LGTM otherwise!

**Review by krishnaglick — APPROVED** [review]
Tested locally, LGTM.
Feels a little strange that Light does something different on Inputs vs Textareas.

**Comment by edmundito** [comment]
@krishnaglick the design of these two needs to be revisited so that they're consistent and match Figma. For now, the goal was to do a bit of cleanup.

**Pull request merged**